File: Data Structures & Algorithms/find-lucky-integer-in-an-array/submission-2.py
from collections import Counter
from typing import List

class Solution:
    def findLucky(self, arr: List[int]) -> int:
        counts = Counter(arr)
        
        lucky = -1
        
        for num, frequency in counts.items():
            if num == frequency:
                lucky = max(lucky, num)
                
        return lucky

File: Data Structures & Algorithms/house-robber/submission-2.cpp
class Solution {
public:
    vector<int> dp;
    int rob(vector<int>& nums) {
        int n = nums.size();
        if(n == 1) return nums[0];
        dp.resize(n, -1);
        dp[n-1] = nums[n-1];
        dp[n-2] = max(nums[n-1], nums[n-2]);
        for(int i = n-3; i>=0; i--) {
            dp[i] = max(nums[i] + dp[i+2], dp[i+1]);
        }
        return dp[0];
    }
};
